plugins/discovery-ec2/src/main/java/org/elasticsearch/discovery/ec2/AwsEc2ServiceImpl.java

```diff
@@ -49,20 +51,50 @@
     private AmazonEC2 buildClient(Ec2ClientSettings clientSettings) {
         final AWSCredentialsProvider credentials = buildCredentials(logger, clientSettings);
         final ClientConfiguration configuration = buildConfiguration(logger, clientSettings);
-        final AmazonEC2 client = buildClient(credentials, configuration);
+
+        final AwsClientBuilder.EndpointConfiguration endpointConfiguration;
         if (Strings.hasText(clientSettings.endpoint)) {
             logger.debug("using explicit ec2 endpoint [{}]", clientSettings.endpoint);
-            client.setEndpoint(clientSettings.endpoint);
+            endpointConfiguration = new AwsClientBuilder.EndpointConfiguration(clientSettings.endpoint,
+                buildRegion(clientSettings.endpoint));
+        } else {
+            logger.debug("No endpoint defined. Using default endpoint [ec2.us-east-1.amazonaws.com].");
+            endpointConfiguration = null;
         }
+
+        final AmazonEC2 client = buildClient(credentials, configuration, endpointConfiguration);
         return client;
     }
 
     // proxy for testing
-    AmazonEC2 buildClient(AWSCredentialsProvider credentials, ClientConfiguration configuration) {
-        final AmazonEC2 client = new AmazonEC2Client(credentials, configuration);
+    AmazonEC2 buildClient(AWSCredentialsProvider credentials, ClientConfiguration configuration,
+                          AwsClientBuilder.EndpointConfiguration endpointConfiguration) {
+
+        AmazonEC2ClientBuilder builder = AmazonEC2ClientBuilder.standard()
+            .withCredentials(credentials)
+            .withClientConfiguration(configuration);
+        if (endpointConfiguration != null) {
+            builder.setEndpointConfiguration(endpointConfiguration);
+        }
+        final AmazonEC2 client = builder.build();
         return client;
     }
 
+    // Package private for tests
+    static String buildRegion(String endpoint) {
```

Comment:
We should get the region from the endpoint or if endpoint is not set, simply return null. Ie. we do not really need to lookup the region using the `InstanceMetadataRegionProvider` since the SDK does that by itself when using the client builder.